src/utils.ts

```diff
@@ -1277,3 +1277,21 @@ export function parseUnsignedInteger(value: unknown): number | null {
 
   return parsedInt != null && parsedInt >= 0 ? parsedInt : null;
 }
+
+/**
+ * Determines whether a provided address matches the provided parent domain.
+ *
+ * If a DNS server were to become compromised SRV records would still need to
+ * advertise addresses that are under the same domain as the srvHost.
+ *
+ * @param address - The address to check against a domain
+ * @param srvHost - The domain to check the provided address against
+ * @returns Whether the provided address matches the parent domain
+ */
+export function matchesParentDomain(address: string, srvHost: string): boolean {
+  const regex = /^.*?\./;
+  const srvAddress = address.endsWith('.') ? address.slice(0, address.length - 1) : address;
+  const srv = `.${srvAddress.replace(regex, '')}`;
+  const parent = `.${srvHost.replace(regex, '')}`;
```

Comment:
Why are we not applying the same logic of stripping a trailing `.` to `srvHost`? Should make sense to allow trailing dots there as well if they’re already handled elsewhere too. (We had a [MONGOSH ticket](https://jira.mongodb.org/browse/MONGOSH-1156) about this a while ago that got resolved on the client side, but I think technically that still stands as a feature request.)